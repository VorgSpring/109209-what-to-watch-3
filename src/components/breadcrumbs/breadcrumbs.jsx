import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {getFilmPathLink} from '../../helpers/get-links/get-links';

export const Breadcrumbs = ({id, name}) => (
  <nav className='breadcrumbs'>
    <ul className='breadcrumbs__list'>
      <li className='breadcrumbs__item'>
        <Link
          to={getFilmPathLink(id)}
          className='breadcrumbs__link'
        >
          {name}
        </Link>
      </li>
      <li className='breadcrumbs__item'>
        <a className='breadcrumbs__link'>Add review</a>
      </li>
    </ul>
  </nav>
);

Breadcrumbs.defaultProps = {
  id: null,
  name: ``
};

Breadcrumbs.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string
};
