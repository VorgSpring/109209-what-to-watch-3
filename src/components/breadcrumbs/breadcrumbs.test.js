import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {Breadcrumbs} from './breadcrumbs.jsx';
import renderer from 'react-test-renderer';

describe(`Breadcrumbs`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Breadcrumbs
            id={13}
            name='blah'
          />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
