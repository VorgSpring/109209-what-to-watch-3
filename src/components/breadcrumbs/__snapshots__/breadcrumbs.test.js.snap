// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should renders correctly 1`] = `
<nav
  className="breadcrumbs"
>
  <ul
    className="breadcrumbs__list"
  >
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/film/13"
        onClick={[Function]}
      >
        blah
      </a>
    </li>
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
      >
        Add review
      </a>
    </li>
  </ul>
</nav>
`;
